packages/connect-wallet-modal/src/components/WalletsModal/WalletsModal.tsx

```diff
@@ -27,113 +21,36 @@ export function WalletsModal(props: WalletsModalProps) {
-  }, [setTermsChecked]);
+  const { currentModal, closeModal, forceCloseAllModals, termsChecked } =
+    useReefKnotModal();
 
   const termsProps: WalletModalConnectTermsProps = {
-    onChange: handleTermsToggle,
-    checked: termsChecked,
     termsLink: termsLink || 'https://lido.fi/terms-of-use',
     privacyNoticeLink: privacyNoticeLink || 'https://lido.fi/privacy-notice',
     metrics,
   };
 
-  const [requirementsVisible, setRequirementsVisible] = useState(false);
-  const [requirementsData, setRequirementsData] = useState<RequirementsData>(
-    {},
-  );
-
-  const setRequirements = useCallback(
-    (isVisible: boolean, reqData: RequirementsData = {}) => {
-      setRequirementsVisible(isVisible);
-      setRequirementsData(reqData);
-    },
-    [],
-  );
-
-  // pass-into function is cheap, so we're losing performance on useCallback here
-  const hideRequirements = () => {
-    setRequirements(false);
-  };
-
-  const [ledgerScreenVisible, setLedgerScreenVisible] = useState(false);
-  const hideLedgerScreen = () => {
-    setLedgerScreenVisible(false);
-  };
-
-  const buttonsCommonProps: ButtonsCommonProps = {
-    disabled: !termsChecked,
-    onConnect: onClose,
-    shouldInvertWalletIcon,
-    setRequirements,
-    setLedgerScreenVisible,
-    metrics,
-  };
-
-  const handleClose = onClose || NOOP;
-
-  const { icon: reqIcon, title: reqTitle, text: reqText } = requirementsData;
-
-  const { acceptTermsModal } = useContext(AcceptTermsModalContext);
-
-  // do not try to render the modal in case of SSR
-  if (typeof window !== 'undefined') {
-    // A conflict is detected or some other requirement must be fulfilled to connect
-    if (requirementsVisible) {
+  const handleCloseSuccess = () => closeModal({ success: true });
+  const handleCloseReject = () => closeModal({ success: false });
+  const handleExit = () => forceCloseAllModals();
+
+  switch (currentModal?.type) {
+    case 'wallet': {
+      const buttonsCommonProps: ButtonsCommonProps = {
+        disabled: !termsChecked,
+        onConnect: handleCloseSuccess,
+        shouldInvertWalletIcon,
+        metrics,
+      };
       return (
         <Modal
-          {...props} // the props are overridden here on purpose
-          onClose={handleClose}
-          onBack={hideRequirements}
-          onExited={hideRequirements}
-          center
-          title={reqTitle}
-          subtitle={reqText}
-          titleIcon={reqIcon}
-        />
-      );
-    }
-
-    if (ledgerScreenVisible) {
-      return (
-        <LedgerModal
-          {...props} // the props are overridden here on purpose
-          onClose={handleClose}
-          onBack={hideLedgerScreen}
-          onExited={hideLedgerScreen}
-          metrics={buttonsCommonProps.metrics}
-        />
-      );
-    }
-
-    if (acceptTermsModal?.isVisible) {
-      return (
-        <AcceptTermsModal
+          {...passedDownProps} // the props can be overridden by a library user
```

Comment:
The new comment is wrong. My comment was about the fact, that a library user CAN'T override such props as `open`, `center`, etc. and it is done on purpose.